lib/models/device.coffee

```diff
@@ -971,6 +985,159 @@ getDeviceModel = (deps, opts) ->
 		.catch(notFoundResponse, treatAsMissingDevice(uuidOrId))
 		.asCallback(callback)
 
+	###*
+	# @summary Start service on device
+	# @name startService
+	# @public
+	# @function
+	# @memberof resin.models.device
+	#
+	# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+	# @param {Number} imageId - id of the image to start
```

Comment:
After reading `Start service on device`, I would expect this to be the service ID :confused: 
Not-for-now: Wouldn't it be great if we could handle the service name as well?
I guess that getting the currently running image from the service name & deviceId isn't straightforward, but I imagine that the consumers might need to do this on their own.

Reply:
It's not that easy. The main problem is that it's possible to have two images from the same service running on one device at the same time, if you're using [handovers](https://docs.resin.io/learn/deploy/release-strategy/update-strategies/#hand-over), or plausibly if something goes wrong somewhere or something. That's important because it's a risky case, and it's a time when you're mostly likely to want to manually kill one of your containers (because your handover has got stuck).

This model is also a _slightly_ nicer in that it makes it hard to hit race conditions, where you try to stop one image while another is downloading and end up stopping a different one, because the supervisor is changing things at the same time you're talking to it. You can avoid that if you specify image ids, but if you can only use services then there's nothing you can do to be more specific.

All that said, I can see it's still not super convenient. What do you think about `startService` + `startServiceByName`? The 2nd would do what you're suggesting (and throw an error if it's ambiguous), but we'd also have the 1st option so you can disambiguate when it matters.

Reply:
> All that said, I can see it's still not super convenient. What do you think about startService + startServiceByName? The 2nd would do what you're suggesting (and throw an error if it's ambiguous), but we'd also have the 1st option so you can disambiguate when it matters.

Sounds great :+1:  . I also don't think that this need to block this PR. We could do it in a separate minor release.

Reply:
Created an issue for later: https://github.com/resin-io/resin-sdk/issues/512